featuretools/primitives/standard/transform/time_series/expanding/expanding_count.py

```diff
@@ -0,0 +1,91 @@
+            and it will indicate a length of time between a target instance and the beginning of its window.
+            Defaults to 1.
+        min_periods (int, optional): Minimum number of observations required for performing calculations
+            over the window. min_periods must always be less than or equal to the length of the series. (TODO: Check? Gap?)
+            When window_length is an offset alias string, this limitation does not exist,
+            but care should be taken to not choose a min_periods that will always be larger than the number of
+            observations in the series.
+            Defaults to 1.
+
+    Note:
+        Only offset aliases with fixed frequencies can be used when defining gap and h.
+        This means that aliases such as `M` or `W` cannot be used, as they can indicate different
+        numbers of days. ('M', because different months have different numbers of days;
+        'W' because week will indicate a certain day of the week, like W-Wed, so that will
+        indicate a different number of days depending on the anchoring date.)
+
+    Note:
+        When using an offset alias to define `gap`, an offset alias must also be used to define `window_length`.
+        This limitation does not exist when using an offset alias to define `window_length`. In fact,
+        if the data has a uniform sampling frequency, it is preferable to use a numeric `gap` as it is more
+        efficient.
+
+
+    Examples:
+        >>> import pandas as pd
+        >>> expanding_count = ExpandingCount()
+        >>> times = pd.date_range(start='2019-01-01', freq='1min', periods=5)
+        >>> expanding_count(times).tolist()
+        [0.0, 1.0, 2.0, 3.0, 4.0]
+
+        We can also control the gap before the expanding calculation.
+
+        >>> import pandas as pd
+        >>> expanding_count = ExpandingCount(gap=0)
+        >>> times = pd.date_range(start='2019-01-01', freq='1min', periods=5)
+        >>> expanding_count(times).tolist()
+        [1.0, 2.0, 3.0, 4.0, 5.0]
+
+        We can also control the minimum number of periods required for the rolling calculation.
+
+        >>> import pandas as pd
+        >>> expanding_count = ExpandingCount(min_periods=3)
+        >>> times = pd.date_range(start='2019-01-01', freq='1min', periods=5)
+        >>> expanding_count(times).tolist()
+        [nan, nan, 2.0, 3.0, 4.0]
+    """
+
+    name = "expanding_count"
+    input_types = [ColumnSchema(logical_type=Datetime, semantic_tags={"time_index"})]
+    return_type = ColumnSchema(logical_type=Double, semantic_tags={"numeric"})
+    uses_full_dataframe = True
+
+    def __init__(self, gap=1, min_periods=0):
+        self.gap = gap
+        self.min_periods = min_periods
+
+    def get_function(self):
+        def expanding_count(datetime):
+            x = pd.Series(1, index=datetime)
+            if isinstance(self.gap, int):
+                x = x.shift(self.gap)
+            else:
```

Comment:
I think the non-integer gap is responsible for much of the complexity in the rolling primitives, but I'm not sure how much benefit there is to the user.